rasterio/rpc.py

```diff
@@ -93,11 +93,16 @@ def from_gdal(cls, rpcs):
         out = {}
 
         for key, val in rpcs.items():
-            vals = val.split()
+            vals = []
```

Comment:
@ungarj I would like to see an approach that is more like GDAL's. That is to say, some RPC items are supposed to be one float and some are supposed to be 20 floats. For the first category, we do `out[key] = float(val.split()[0])`. For the second category we do `out[key] = [float(v) for v in val.split()[:20]]`.

Reply:
@sgillies I.e. limit the values to 20 like this? (see ceed75caf683f94e208cc56cc2b31db99daab72c)

Reply:
@ungarj even earlier than that. There are exactly 4 keys which can have 20 values: line_num_coeff, line_den_coeff, samp_num_coeff, samp_den_coeff. So, something like 

```
for key, val in rpcs.items():
    # Four items have 20 floats in their values.
    if key in {"line_num_coeff", "line_den_coeff", "samp_num_coeff", "samp_den_coeff"}:
        out[key] = [float(v) for v in val.split()[:20]]
    # All other items have one float in their values but might also contain non-conforming extra text.
    else:
        out[key] = float(val.split()[0])
```

Reply:
Yes, that makes perfect sense. Sorry for not seeing that earlier.

I basically inserted your code now, except changing the keys to uppercase.